Issue #232: migration to sonarqube 7.9 LTS

romani opened the pull request:
Issue #232

romani commented:
wercker is failed because in docker container current user is "sonarqube" but werkcer expect to be executed under root (creation of folder `/pipeline` is not only his requirements)
the same problem as https://github.com/SonarSource/docker-sonarqube/issues/230 , we need to use 
https://github.com/sonarsource/orchestrator

romani commented:
@muhlba91 , I pushed my temp changes on how to move integration testing to `it`.
Ones it completed we can completely disable wercker for this build, we will find good use of it later on ...
we can keep it on old docker image and disable all actions .....

Please help to continue update to 7.9.

muhlba91 commented:
@romani - thank you very much. :)

Currently, I am on holidays until 7th January but will have a look at it after I‘m back and continue the upgrade.

muhlba91 commented:
@romani i have updated the dependencies and Travis runs through. The problem was that some libraries implicitly used a newer `slf4j` version whereas we transitively loaded an older one. Adding the correct dependencies and updating `slf4j` fixed the issues.
Please review. :)

Wercker is supposed to not run through, right?

rnveach commented on src/it/java/org/checkstyle/plugins/sonar/RunPluginTest.java:
```diff
@@ -0,0 +1,163 @@
+ * and from
+ * https://github.com/SonarSource/sonar-jacoco/blob/master/its/src/test/java/org/sonar/plugins/jacoco/its/JacocoTest.java
+ */
+class RunPluginTest {
```
This file is missing the header and line width restriction.

romani replied:
moved to #269

romani commented:
> Wercker is supposed to not run through, right?

yes, I pushed a commit to make wercker do simple build, in future we might use it for something more useful.

romani commented on src/it/java/org/checkstyle/plugins/sonar/RunPluginTest.java:
```diff
@@ -0,0 +1,163 @@
+    mavenBuild.setProperty("sonar.projectKey", projectKey);
+    return mavenBuild;
+  }
+}
```
\n

romani replied:
resolved.

romani commented on src/it/java/org/checkstyle/plugins/sonar/RunPluginTest.java:
```diff
@@ -0,0 +1,163 @@
+    MavenBuild build = test_project("com.puppycrows.tools:checkstyle", "src/");
+    executeBuildWithCommonProperties(build, projectName, false);
+
+        List<String> metricKeys = Arrays.asList("line_coverage", "lines_to_cover",
```
please clear code a bit with indentation ...
It is kind of strange why checksyle validation does not detect this ....

romani replied:
chekcstyle is updated - https://github.com/checkstyle/sonar-checkstyle/pull/260

rnveach replied:
@romani Checkstyle is not running on these files since they are in `it` folder. Checkstyle maven run has to be updated with this new folder.

romani replied:
thanks a lot !! 
I finally found root reason - it is resolved at #260 .

rnveach replied:
@romani It is not resolved as CI still allows the same code in.

romani replied:
will be addressed at #269

romani commented on src/it/java/org/checkstyle/plugins/sonar/RunPluginTest.java:
```diff
@@ -0,0 +1,163 @@
+import org.sonarqube.ws.client.measure.ComponentWsRequest;
+
+/**
+ * a lot of code from https://github.com/SonarSource/sonar-java/blob/5aaaeba90cd85a40391706ad2f806d6a156f80cb/its/ruling/src/test/java/org/sonar/java/it/JavaRulingTest.java
```
we do not need such comment any more, I used it as placeholder  for links where I found good examples.

romani replied:
removed

romani commented:
@muhlba91 , please merge https://github.com/checkstyle/sonar-checkstyle/pull/260
and rebase and squash this branch to make it ready for merge.

romani commented:
@muhlba91 , there are bunch of noise kin tests logs - https://travis-ci.org/checkstyle/sonar-checkstyle/jobs/636480400#L301

is it possible to resolve them ?  can be postponed till next PR.

muhlba91 commented:
@romani i have added a `logback.xml` file for the tests which only logs messages above the `error` level during the tests. this removes some of the noise at least.

we have a problem with TeamCity and Checkstyle in `CheckstyleConfiguration::getTargetXmlReport` (L103/L104): https://github.com/checkstyle/sonar-checkstyle/blob/master/src/main/java/org/sonar/plugins/checkstyle/CheckstyleConfiguration.java#L104
since this lambda only returns one value, TC expects it like this (which adheres to the common standard):
```
.map(aBoolean -> new File(fileSystem.workDir(), "checkstyle-result.xml"))
```
Checkstyle, on the other hand, expects it to be as it is at the moment.
as those 2 expect exactly the opposite, one always has to fail (TC now).
what has precendence here - CS or TC? can we adapt CS to allow such lambdas?

i hav also auto-formatted the `RunPluginTest` using IntelliJ to clear up indentations.

romani commented:
I did some updates and pushed changes with force to remote ....
such 'it' doesnot  not looks like executing at all, It is not compile in IDEA.

romani commented:
Maven do not see `it` sources at all, so they are not compilable and Checkstyle does not see them also, it it takes sources from maven.

muhlba91 commented:
@romani - is there anything still missing now or are we ready to merge? :)

romani commented:
Yes, RunPluginTest is just uncompilable text file. It is not compiled, no prove that it works.

muhlba91 commented:
what‘s the reason actually to place it in the it folder and not to the other tests?

romani commented:
`ut` suppose to be executed when classes are just compiled but not bundled to JAR
`it` should be executed when maven packaged JAR, so we tell Sonar plugin to use JAR - integration test (as it was in wercker - no fake, complete emulation of user JAR usage).

muhlba91 commented:
hmmm... ok, I have never worked with such kind of tests.

however, locally i have added this folder as a test source folder so that it gets executed with all other tests for the moment and realized 2 things:
- the used API is from sonar 6.7, 7.9 changed quite a bit
- there are quite some artifacts from jacoco

i tried to clean it up but am somehow stuck in its purpose and workflow.
for example: there's this constant `FILE_KEY` pointing to a non-existing file, `metricKeys` used look like from jacoco, the `run` method should return nothing but it returns something (uncompilable yet due to API changes), `preparePreoject` was not copied but has two different version in the linked files.

i really need some help here @romani to clean this one up properly with the desired outcome, measures, ... i have pushed a temporary commit s.t. you can run the tests now. at the moment it fails due to non-existing files, ...

muhlba91 commented:
@romani there seems to be a problem now.

i got the file running and executing but i had to use jdk 11 since SQ 7.9 requires it.

however, due to proxy issues i wasn't able to test it locally, on travis it fails due to it detecting it's either a branch or pull request and the used SQ edition is community.

i have changed the edition to developer but for getting this working, we need an artifactory api key as per https://github.com/SonarSource/orchestrator#configuration
hence, a configuration file containing
```
orchestrator.artifactory.apiKey=<KEY>
```
has to be created and linked through `-Dorchestrator.configUrl=file:///path/to/orchestrator.properties` or `ORCHESTRATOR_CONFIG_URL`.
can you please facilitate this in the travis configuration as we shouldn't commit api keys in git, obviously? ;)

rnveach commented:
> i had to use jdk 11 since SQ 7.9 requires it.

Just wanted to confirm. This PR, when merged, will require all sonar installations to use JDK 11 to use checkstyle sonar from now on?

muhlba91 commented:
> > i had to use jdk 11 since SQ 7.9 requires it.
> 
> This PR, when merged, will require all sonar installations to use JDK 11 to use checkstyle sonar from now on?

SQ 7.9 definitely requires Java 11 to run. I don't see any blockers why we shouldn't switch entirely to JDK 11 unless you see some? Also, it will be easier for developers to understand that JDK 11 is actually a requirement because otherwise the integration test will fail.

Anyhow, for the moment, I have just updated the affected Travis build (`verify`) to use `openjdk11` but left the others at `openjdk8´. As long as we don't run the tests for releases, we could, theoretically, still use JDK 8 although I believe it will lead to confusion regarding different requirements.

rnveach commented:
> I don't see any blockers why we shouldn't switch entirely to JDK 11 unless you see some?

I leave this up for @romani to decide on. He is for upgrading to new versions. When we updated Checkstyle to JDK 8, there was some who didn't like that we were forcing them to upgrade and held back.

romani commented:
It is completely ok to upgrade plugin to same jdk as host system require as minimal.
Please migrate all to jdk11

romani commented:
@muhlba91 , I resolved Travis 2 builds issue, and I rebased and squashed all commit and pushed with force.
Please take branch from remote, before any actions.

romani commented:
> please put jdk version requilrements in pom.xm

done.

romani commented:
I converted to use community edition:
```
-                .setEdition(Edition.DEVELOPER)
+                .setEdition(Edition.COMMUNITY)
```
So it starts working .... hanging.... after attaching debug to see what it is doing .... it is downloading sonarqube (200Mb file) ... to `target/temp-downloads/sonar-application-7.9.2.zip`

```
$ URL="https://repox.jfrog.io/repox/sonarsource/org/sonarsource/sonarqube/sonar-application/7.9.2/sonar-application-7.9.2.zip"
$ curl -sI $URL | grep -i Content-Length
Content-Length: 209517312
```

my Firefox estimated download of such url in  40min ...

romani commented:
I recommend to download zip manually and  use my last commit ... now it is running:
```
$ mvn verify

.....

[INFO] Building jar: /home/rivanov/java/github/checkstyle/sonar-checkstyle/target/checkstyle-sonar-plugin-4.28-SNAPSHOT.jar
[INFO]  already added, skipping
[INFO] 
[INFO] --- maven-failsafe-plugin:2.22.2:integration-test (default) @ checkstyle-sonar-plugin ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.checkstyle.plugins.sonar.RunPluginTest
> 15:06:20.664 [main] WARN org.sonar.application.config.JdbcSettings - Both 'JDBC_EMBEDDED_PORT' and 'JDBC_URL' properties are set. The value of property 'JDBC_URL' ('jdbc:h2:tcp://127.0.0.1:33622/sonar;USER=sonar;PASSWORD=sonar') is not consistent with the value of property 'JDBC_EMBEDDED_PORT' ('33622'). The value of property 'JDBC_URL' will be ignored and value 'jdbc:h2:tcp://127.0.0.1:33622/sonar' will be used instead. To remove this warning, either remove property 'JDBC_URL' if your intent was to use the embedded H2 database, otherwise remove property 'JDBC_EMBEDDED_PORT'.
> 2020.01.17 15:06:20 INFO  app[][o.s.a.AppFileSystem] Cleaning or creating temp directory /home/rivanov/java/github/checkstyle/sonar-checkstyle/target/1/sonarqube-7.9.2/temp
> 2020.01.17 15:06:20 INFO  app[][o.s.a.es.EsSettings] Elasticsearch listening on /127.0.0.1:39590
> 2020.01.17 15:06:20 INFO  app[][o.s.a.ProcessLauncherImpl] Launch process[[key='es', ipcIndex=1, logFilenamePrefix=es]] from [/home/rivanov/java/github/checkstyle/sonar-checkstyle/target/1/sonarqube-7.9.2/elasticsearch]: /home/rivanov/java/github/checkstyle/sonar-checkstyle/target/1/sonarqube-7.9.2/elasticsearch/bin/elasticsearch
> 2020.01.17 15:06:20 INFO  app[][o.s.a.SchedulerImpl] Waiting for Elasticsearch to be up and running
> OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
> 2020.01.17 15:06:21 INFO  app[][o.e.p.PluginsService] no modules loaded
> 2020.01.17 15:06:21 INFO  app[][o.e.p.PluginsService] loaded plugin [org.elasticsearch.transport.Netty4Plugin]
> 2020.01.17 15:06:28 INFO  app[][o.s.a.SchedulerImpl] Process[es] is up
> 2020.01.17 15:06:28 INFO  app[][o.s.a.ProcessLauncherImpl] Launch process[[key='web', ipcIndex=2, logFilenamePrefix=web]] from [/home/rivanov/java/github/checkstyle/sonar-checkstyle/target/1/sonarqube-7.9.2]: /opt/jvm/openjdk-12-32/bin/java -Djava.awt.headless=true -Dfile.encoding=UTF-8 -Djava.io.tmpdir=/home/rivanov/java/github/checkstyle/sonar-checkstyle/target/1/sonarqube-7.9.2/temp --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED -Xmx1G -Djava.net.preferIPv4Stack=true -Djava.security.egd=file:/dev/./urandom -Dhttp.nonProxyHosts=localhost|127.*|[::1] -cp ./lib/common/*:/home/rivanov/java/github/checkstyle/sonar-checkstyle/target/1/sonarqube-7.9.2/lib/jdbc/h2/h2-1.3.176.jar org.sonar.server.app.WebServer /home/rivanov/java/github/checkstyle/sonar-checkstyle/target/1/sonarqube-7.9.2/temp/sq-process13799996275091292011properties
> 2020.01.17 15:06:29 INFO  web[][o.s.p.ProcessEntryPoint] Starting web
> 2020.01.17 15:06:29 INFO  web[][o.a.t.u.n.NioSelectorPool] Using a shared selector for servlet write/read
> 2020.01.17 15:06:30 INFO  web[][o.e.p.PluginsService] no modules loaded
> 2020.01.17 15:06:30 INFO  web[][o.e.p.PluginsService] loaded plugin [org.elasticsearch.join.ParentJoinPlugin]
> 2020.01.17 15:06:30 INFO  web[][o.e.p.PluginsService] loaded plugin [org.elasticsearch.percolator.PercolatorPlugin]
> 2020.01.17 15:06:30 INFO  web[][o.e.p.PluginsService] loaded plugin [org.elasticsearch.transport.Netty4Plugin]
> 2020.01.17 15:06:31 INFO  web[][o.s.s.e.EsClientProvider] Connected to local Elasticsearch: [127.0.0.1:39590]
> 2020.01.17 15:06:31 INFO  web[][o.s.s.p.LogServerVersion] SonarQube Server / 7.9.2.30863 / cd30425aa0b4e62c39b57c81d64be16e365f6d83
> 2020.01.17 15:06:31 INFO  web[][o.s.s.p.d.EmbeddedDatabase] Starting embedded database on port 33622 with url jdbc:h2:tcp://127.0.0.1:33622/sonar
> 2020.01.17 15:06:31 INFO  web[][o.s.s.p.d.EmbeddedDatabase] Embedded database started. Data stored in: /home/rivanov/java/github/checkstyle/sonar-checkstyle/target/1/sonarqube-7.9.2/data
> 2020.01.17 15:06:31 INFO  web[][o.sonar.db.Database] Create JDBC data source for jdbc:h2:tcp://127.0.0.1:33622/sonar
> 2020.01.17 15:06:31 WARN  web[][o.s.db.dialect.H2] H2 database should be used for evaluation purpose only.
> 2020.01.17 15:06:32 INFO  web[][o.s.s.p.ServerFileSystemImpl] SonarQube home: /home/rivanov/java/github/checkstyle/sonar-checkstyle/target/1/sonarqube-7.9.2
> 2020.01.17 15:06:32 INFO  web[][o.s.s.u.SystemPasscodeImpl] System authentication by passcode is disabled
> 2020.01.17 15:06:32 INFO  web[][o.s.s.p.d.m.h.MigrationHistoryTableImpl] Creating table schema_migrations
> 2020.01.17 15:06:32 INFO  web[][o.s.s.p.ServerPluginRepository] Plugin LITS [lits] installed
> 2020.01.17 15:06:32 INFO  web[][o.s.s.p.ServerPluginRepository] Plugin Orchestrator :: Reset data [resetdata] installed
> 2020.01.17 15:06:32 INFO  web[][o.s.s.p.ServerPluginRepository] Plugin Checkstyle [checkstyle] installed
> 2020.01.17 15:06:32 INFO  web[][o.s.s.p.ServerPluginRepository] Deploy plugin Checkstyle / 4.28-SNAPSHOT / 
> 2020.01.17 15:06:32 INFO  web[][o.s.s.p.ServerPluginRepository] Deploy plugin LITS / 0.8.0.1209 / 48bf9f2d3bbd5131dddefc41c093f9ad21235c02
> 2020.01.17 15:06:32 INFO  web[][o.s.s.p.ServerPluginRepository] Deploy plugin Orchestrator :: Reset data / 1.0-SNAPSHOT
> 2020.01.17 15:06:32 INFO  web[][o.s.s.p.d.m.AutoDbMigration] Automatically perform DB migration on fresh install
> 2020.01.17 15:06:33 INFO  web[][o.s.s.p.Platform] Database has been automatically updated
> 2020.01.17 15:06:33 INFO  web[][o.s.s.p.w.MasterServletFilter] Initializing servlet filter org.sonar.server.ws.WebServiceFilter@76d8bcef [pattern=UrlPattern{inclusions=[/api/system/migrate_db.*, ...], exclusions=[/api/properties*, ...]}]
> 2020.01.17 15:06:33 INFO  web[][o.s.s.a.EmbeddedTomcat] HTTP connector enabled on port 38494
> 2020.01.17 15:06:33 INFO  web[][o.s.s.p.UpdateCenterClient] Update center: https://update.sonarsource.org/update-center.properties (no proxy)
> 2020.01.17 15:06:33 INFO  web[][o.s.s.e.IndexCreator] Create index [metadatas]
> 2020.01.17 15:06:34 INFO  web[][o.s.s.e.IndexCreator] Create type metadatas/metadata
> 2020.01.17 15:06:34 INFO  web[][o.s.s.e.IndexCreator] Create index [components]
> 2020.01.17 15:06:35 INFO  web[][o.s.s.e.IndexCreator] Create type components/auth
> 2020.01.17 15:06:35 INFO  web[][o.s.s.e.IndexCreator] Create index [projectmeasures]
> 2020.01.17 15:06:36 INFO  web[][o.s.s.e.IndexCreator] Create type projectmeasures/auth
> 2020.01.17 15:06:36 INFO  web[][o.s.s.e.IndexCreator] Create index [rules]
> 2020.01.17 15:06:37 INFO  web[][o.s.s.e.IndexCreator] Create type rules/rule
> 2020.01.17 15:06:37 INFO  web[][o.s.s.e.IndexCreator] Create index [issues]
> 2020.01.17 15:06:38 INFO  web[][o.s.s.e.IndexCreator] Create type issues/auth
> 2020.01.17 15:06:38 INFO  web[][o.s.s.e.IndexCreator] Create index [users]
> 2020.01.17 15:06:38 INFO  web[][o.s.s.e.IndexCreator] Create type users/user
> 2020.01.17 15:06:38 INFO  web[][o.s.s.e.IndexCreator] Create index [views]
> 2020.01.17 15:06:39 INFO  web[][o.s.s.e.IndexCreator] Create type views/view
> 2020.01.17 15:06:39 INFO  web[][o.s.s.s.LogServerId] Server ID: E37E2F91-AW-1whUYjgZ2x9hd-JK9
> 2020.01.17 15:06:39 WARN  web[][o.s.a.s.w.WebService$Action] The response example is not set on action api/plugins/download
> 2020.01.17 15:06:39 WARN  web[][o.s.a.s.w.WebService$Action] The response example is not set on action api/permissions/search_templates
> 2020.01.17 15:06:39 INFO  web[][o.s.s.t.TelemetryDaemon] Sharing of SonarQube statistics is enabled.
> 2020.01.17 15:06:39 INFO  web[][o.s.s.n.NotificationDaemon] Notification service started (delay 60 sec.)
> 2020.01.17 15:06:39 INFO  web[][o.s.s.s.GeneratePluginIndex] Generate scanner plugin index
> 2020.01.17 15:06:39 INFO  web[][o.s.s.s.RegisterPlugins] Register plugins
> 2020.01.17 15:06:39 INFO  web[][o.s.s.s.RegisterMetrics] Register metrics
> 2020.01.17 15:06:39 INFO  web[][o.s.s.q.RegisterQualityGates] Built-in quality gate's conditions of [Sonar way] has been updated
> 2020.01.17 15:06:39 INFO  web[][o.s.s.r.RegisterRules] Register rules
> 2020.01.17 15:06:39 INFO  web[][o.s.s.q.BuiltInQProfileRepositoryImpl] Load quality profiles
> 2020.01.17 15:06:39 INFO  web[][o.s.s.s.RegisterPermissionTemplates] Register permission templates
> 2020.01.17 15:06:39 INFO  web[][o.s.s.s.RenameDeprecatedPropertyKeys] Rename deprecated property keys
> 2020.01.17 15:06:39 INFO  web[][o.s.s.p.w.MasterServletFilter] Initializing servlet filter org.sonar.server.ws.WebServiceFilter@50623c8a [pattern=UrlPattern{inclusions=[/api/issues/delete_comment.*, ...], exclusions=[/api/properties*, ...]}]
> 2020.01.17 15:06:39 INFO  web[][o.s.s.p.w.MasterServletFilter] Initializing servlet filter org.sonar.server.ws.DeprecatedPropertiesWsFilter@5b25e324 [pattern=UrlPattern{inclusions=[/api/properties/*], exclusions=[]}]
> 2020.01.17 15:06:39 INFO  web[][o.s.s.p.w.MasterServletFilter] Initializing servlet filter org.sonar.server.ws.WebServiceReroutingFilter@4450d0bb [pattern=UrlPattern{inclusions=[/api/components/bulk_update_key, ...], exclusions=[]}]
> 2020.01.17 15:06:39 INFO  web[][o.s.s.p.w.MasterServletFilter] Initializing servlet filter org.sonar.server.authentication.InitFilter@43db1dd2 [pattern=UrlPattern{inclusions=[/sessions/init/*], exclusions=[]}]
> 2020.01.17 15:06:39 INFO  web[][o.s.s.p.w.MasterServletFilter] Initializing servlet filter org.sonar.server.authentication.OAuth2CallbackFilter@c34684d [pattern=UrlPattern{inclusions=[/oauth2/callback/*], exclusions=[]}]
> 2020.01.17 15:06:39 INFO  web[][o.s.s.p.w.MasterServletFilter] Initializing servlet filter org.sonar.server.authentication.ws.LoginAction@664d7f69 [pattern=UrlPattern{inclusions=[/api/authentication/login], exclusions=[]}]
> 2020.01.17 15:06:39 INFO  web[][o.s.s.p.w.MasterServletFilter] Initializing servlet filter org.sonar.server.authentication.ws.LogoutAction@1a85ad00 [pattern=UrlPattern{inclusions=[/api/authentication/logout], exclusions=[]}]
> 2020.01.17 15:06:39 INFO  web[][o.s.s.p.w.MasterServletFilter] Initializing servlet filter org.sonar.server.authentication.ws.ValidateAction@747a1f34 [pattern=UrlPattern{inclusions=[/api/authentication/validate], exclusions=[]}]
> 2020.01.17 15:06:39 INFO  web[][o.s.s.e.IndexerStartupTask] Indexing of type [rules/rule/activeRule] ...
> 2020.01.17 15:06:39 INFO  web[][o.s.s.e.IndexerStartupTask] Indexing of type [rules/rule/activeRule] done | time=99ms
> 2020.01.17 15:06:39 INFO  web[][o.s.s.e.IndexerStartupTask] Indexing of types [rules/rule],[rules/rule/ruleExtension] ...
> 2020.01.17 15:06:39 INFO  web[][o.s.s.e.IndexerStartupTask] Indexing of types [rules/rule],[rules/rule/ruleExtension] done | time=119ms
> 2020.01.17 15:06:39 INFO  web[][o.s.s.e.IndexerStartupTask] Indexing of type [users/user] ...
> 2020.01.17 15:06:40 INFO  web[][o.s.s.e.IndexerStartupTask] Indexing of type [users/user] done | time=340ms
> 2020.01.17 15:06:40 INFO  web[][o.s.s.e.IndexerStartupTask] Indexing of type [components/auth/component] ...
> 2020.01.17 15:06:40 INFO  web[][o.s.s.e.IndexerStartupTask] Indexing of type [components/auth/component] done | time=142ms
> 2020.01.17 15:06:40 INFO  web[][o.s.s.e.IndexerStartupTask] Indexing of type [views/view] ...
> 2020.01.17 15:06:40 INFO  web[][o.s.s.e.IndexerStartupTask] Indexing of type [views/view] done | time=103ms
> 2020.01.17 15:06:40 INFO  web[][o.s.s.e.IndexerStartupTask] Indexing of type [issues/auth/issue] ...
> 2020.01.17 15:06:40 INFO  web[][o.s.s.e.IndexerStartupTask] Indexing of type [issues/auth/issue] done | time=71ms
> 2020.01.17 15:06:40 INFO  web[][o.s.s.e.IndexerStartupTask] Indexing of types [components/auth],[projectmeasures/auth],[issues/auth] ...
> 2020.01.17 15:06:40 INFO  web[][o.s.s.e.IndexerStartupTask] Indexing of types [components/auth],[projectmeasures/auth],[issues/auth] done | time=105ms
> 2020.01.17 15:06:40 INFO  web[][o.s.s.e.IndexerStartupTask] Indexing of type [projectmeasures/auth/projectmeasure] ...
> 2020.01.17 15:06:40 INFO  web[][o.s.s.e.IndexerStartupTask] Indexing of type [projectmeasures/auth/projectmeasure] done | time=72ms
> 2020.01.17 15:06:40 INFO  web[][o.s.s.q.ProjectsInWarningDaemon] Counting number of projects in warning is enabled.
> 2020.01.17 15:06:40 INFO  web[][o.s.s.p.p.PlatformLevelStartup] Running Community Edition
> 2020.01.17 15:06:40 INFO  web[][o.s.s.p.Platform] WebServer is operational
> 2020.01.17 15:06:40 INFO  web[][o.s.s.q.ProjectsInWarningDaemon] Counting number of projects in warning will be disabled as there are no more projects in warning.
> 2020.01.17 15:06:40 INFO  app[][o.s.a.SchedulerImpl] Process[web] is up
> 2020.01.17 15:06:40 INFO  app[][o.s.a.ProcessLauncherImpl] Launch process[[key='ce', ipcIndex=3, logFilenamePrefix=ce]] from [/home/rivanov/java/github/checkstyle/sonar-checkstyle/target/1/sonarqube-7.9.2]: /opt/jvm/openjdk-12-32/bin/java -Djava.awt.headless=true -Dfile.encoding=UTF-8 -Djava.io.tmpdir=/home/rivanov/java/github/checkstyle/sonar-checkstyle/target/1/sonarqube-7.9.2/temp --add-opens=java.base/java.util=ALL-UNNAMED -Xmx512m -Xms128m -XX:+HeapDumpOnOutOfMemoryError -Djava.net.preferIPv4Stack=true -Djava.security.egd=file:/dev/./urandom -Dhttp.nonProxyHosts=localhost|127.*|[::1] -cp ./lib/common/*:/home/rivanov/java/github/checkstyle/sonar-checkstyle/target/1/sonarqube-7.9.2/lib/jdbc/h2/h2-1.3.176.jar org.sonar.ce.app.CeServer /home/rivanov/java/github/checkstyle/sonar-checkstyle/target/1/sonarqube-7.9.2/temp/sq-process5547213144002139566properties
> 2020.01.17 15:06:41 INFO  ce[][o.s.p.ProcessEntryPoint] Starting ce
> 2020.01.17 15:06:41 INFO  ce[][o.s.ce.app.CeServer] Compute Engine starting up...
> 2020.01.17 15:06:41 INFO  ce[][o.e.p.PluginsService] no modules loaded
> 2020.01.17 15:06:41 INFO  ce[][o.e.p.PluginsService] loaded plugin [org.elasticsearch.join.ParentJoinPlugin]
> 2020.01.17 15:06:41 INFO  ce[][o.e.p.PluginsService] loaded plugin [org.elasticsearch.percolator.PercolatorPlugin]
> 2020.01.17 15:06:41 INFO  ce[][o.e.p.PluginsService] loaded plugin [org.elasticsearch.transport.Netty4Plugin]
> 2020.01.17 15:06:42 INFO  ce[][o.s.s.e.EsClientProvider] Connected to local Elasticsearch: [127.0.0.1:39590]
> 2020.01.17 15:06:42 INFO  ce[][o.sonar.db.Database] Create JDBC data source for jdbc:h2:tcp://127.0.0.1:33622/sonar
> 2020.01.17 15:06:42 WARN  ce[][o.s.db.dialect.H2] H2 database should be used for evaluation purpose only.
> 2020.01.17 15:06:43 INFO  ce[][o.s.s.p.ServerFileSystemImpl] SonarQube home: /home/rivanov/java/github/checkstyle/sonar-checkstyle/target/1/sonarqube-7.9.2
> 2020.01.17 15:06:43 INFO  ce[][o.s.c.c.CePluginRepository] Load plugins
> 2020.01.17 15:06:43 INFO  ce[][o.s.c.c.ComputeEngineContainerImpl] Running Community edition
> 2020.01.17 15:06:43 INFO  ce[][o.s.ce.app.CeServer] Compute Engine is operational
> 2020.01.17 15:06:43 INFO  app[][o.s.a.SchedulerImpl] Process[ce] is up
> 2020.01.17 15:06:43 INFO  app[][o.s.a.SchedulerImpl] SonarQube is up
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< com.puppycrows.tools:checkstyle >-------------------
[INFO] Building integration-test-project 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ checkstyle ---
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ checkstyle ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /tmp/junit3135177329354382913/integration-test-project/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ checkstyle ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 1 source file to /tmp/junit3135177329354382913/integration-test-project/target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] Source option 6 is no longer supported. Use 7 or later.
[ERROR] Target option 6 is no longer supported. Use 7 or later.
[INFO] 2 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.772 s
[INFO] Finished at: 2020-01-17T15:06:46-08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project checkstyle: Compilation failure: Compilation failure: 
[ERROR] Source option 6 is no longer supported. Use 7 or later.
[ERROR] Target option 6 is no longer supported. Use 7 or later.
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project checkstyle: Compilation failure
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:567)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.compiler.CompilationFailureException: Compilation failure
    at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:1220)
    at org.apache.maven.plugin.compiler.CompilerMojo.execute (CompilerMojo.java:187)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:567)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< com.puppycrows.tools:checkstyle >-------------------
[INFO] Building integration-test-project 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- sonar-maven-plugin:3.4.1.1168:sonar (default-cli) @ checkstyle ---
[INFO] User cache: /home/rivanov/.sonar/cache
[INFO] SonarQube version: 7.9.2
[INFO] Default locale: "en_US", source code encoding: "UTF-8"
[INFO] Load global settings
[INFO] Load global settings (done) | time=373ms
[INFO] Server id: E37E2F91-AW-1whUYjgZ2x9hd-JK9
[INFO] User cache: /home/rivanov/.sonar/cache
[INFO] Load/download plugins
[INFO] Load plugins index
[INFO] Load plugins index (done) | time=336ms
[INFO] Load/download plugins (done) | time=1317ms
[INFO] Process project properties
[INFO] Project key: com.puppycrows.tools:checkstyle
[INFO] Base dir: /tmp/junit3135177329354382913/integration-test-project
[INFO] Working dir: /tmp/junit3135177329354382913/integration-test-project/target/sonar
[INFO] Load project settings for component key: 'com.puppycrows.tools:checkstyle'
[INFO] Load project settings for component key: 'com.puppycrows.tools:checkstyle' (done) | time=331ms
[INFO] Load quality profiles
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.608 s
[INFO] Finished at: 2020-01-17T15:06:50-08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.4.1.1168:sonar (default-cli) on project checkstyle: No quality profiles have been found, you probably don't have any language plugin installed. -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.4.1.1168:sonar (default-cli) on project checkstyle: No quality profiles have been found, you probably don't have any language plugin installed.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:567)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException: No quality profiles have been found, you probably don't have any language plugin installed.
    at org.sonarsource.scanner.maven.bootstrap.ScannerBootstrapper.execute (ScannerBootstrapper.java:67)
    at org.sonarsource.scanner.maven.SonarQubeMojo.execute (SonarQubeMojo.java:104)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:567)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.sonar.api.utils.MessageException: No quality profiles have been found, you probably don't have any language plugin installed.
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
> 2020.01.17 15:06:51 INFO  app[][o.s.a.SchedulerImpl] Hard stopping SonarQube
> 2020.01.17 15:06:52 INFO  ce[][o.s.p.ProcessEntryPoint] Hard stopping process
> 2020.01.17 15:06:52 INFO  ce[][o.s.ce.app.CeServer] Compute Engine is stopping...
> 2020.01.17 15:06:52 INFO  ce[][o.s.c.t.CeProcessingSchedulerImpl] Hard stopping workers...
> 2020.01.17 15:06:52 INFO  ce[][o.s.ce.app.CeServer] Compute Engine is stopped
> 2020.01.17 15:06:52 INFO  app[][o.s.a.SchedulerImpl] Process[ce] is stopped
> 2020.01.17 15:06:52 INFO  web[][o.s.p.ProcessEntryPoint] Hard stopping process
> 2020.01.17 15:06:52 INFO  web[][o.s.s.n.NotificationDaemon] Notification service stopped
> 2020.01.17 15:06:53 WARN  web[][o.s.p.ProcessEntryPoint$HardStopperThread] Can not stop in 1000ms
> 2020.01.17 15:06:53 WARN  web[][o.s.s.a.EmbeddedTomcat] Failed to stop web server
> org.apache.catalina.LifecycleException: Failed to stop component [StandardServer[-1]]
>   at org.apache.catalina.util.LifecycleBase.stop(LifecycleBase.java:238)
>   at org.apache.catalina.startup.Tomcat.stop(Tomcat.java:437)
>   at org.sonar.server.app.EmbeddedTomcat.terminate(EmbeddedTomcat.java:104)
>   at org.sonar.server.app.WebServer.hardStop(WebServer.java:83)
>   at org.sonar.process.ProcessEntryPoint$HardStopperThread.lambda$new$0(ProcessEntryPoint.java:219)
>   at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
>   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
>   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
>   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
>   at java.base/java.lang.Thread.run(Thread.java:835)
> Caused by: org.apache.catalina.LifecycleException: Failed to stop component [StandardService[Tomcat]]
>   at org.apache.catalina.util.LifecycleBase.stop(LifecycleBase.java:238)
>   at org.apache.catalina.core.StandardServer.stopInternal(StandardServer.java:814)
>   at org.apache.catalina.util.LifecycleBase.stop(LifecycleBase.java:226)
>   ... 9 common frames omitted
> Caused by: org.apache.catalina.LifecycleException: Failed to stop component [StandardEngine[Tomcat]]
>   at org.apache.catalina.util.LifecycleBase.stop(LifecycleBase.java:238)
>   at org.apache.catalina.core.StandardService.stopInternal(StandardService.java:486)
>   at org.apache.catalina.util.LifecycleBase.stop(LifecycleBase.java:226)
>   ... 11 common frames omitted
> Caused by: org.apache.catalina.LifecycleException: A child container failed during stop
>   at org.apache.catalina.core.ContainerBase.stopInternal(ContainerBase.java:1008)
>   at org.apache.catalina.util.LifecycleBase.stop(LifecycleBase.java:226)
>   ... 13 common frames omitted
> 2020.01.17 15:06:53 INFO  web[][o.s.s.p.d.EmbeddedDatabase] Embedded database stopped
> 2020.01.17 15:06:54 INFO  app[][o.s.a.SchedulerImpl] Process[web] is stopped
> 2020.01.17 15:06:54 WARN  app[][o.s.a.p.AbstractManagedProcess] Process exited with exit value [es]: 143
> 2020.01.17 15:06:54 INFO  app[][o.s.a.SchedulerImpl] Process[es] is stopped
> 2020.01.17 15:06:54 INFO  app[][o.s.a.SchedulerImpl] SonarQube is stopped
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 36.698 s <<< FAILURE! - in org.checkstyle.plugins.sonar.RunPluginTest
[ERROR] run(org.checkstyle.plugins.sonar.RunPluginTest)  Time elapsed: 7.437 s  <<< FAILURE!
java.lang.AssertionError: Build failure for project: integration-test-project
  at org.checkstyle.plugins.sonar.RunPluginTest.executeBuildWithCommonProperties(RunPluginTest.java:107)
  at org.checkstyle.plugins.sonar.RunPluginTest.run(RunPluginTest.java:73)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   RunPluginTest.run:73->executeBuildWithCommonProperties:107 Build failure for project: integration-test-project
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0

```

please continue .... I will think on how to cache this zip in CI and skip IT executions by default and only do this in CI.

romani commented:
@muhlba91 , if `it` completion become problematic, we can make migration to 7.9 and manually test it, and make `it` automation as next PR.

muhlba91 commented:
@romani i fixed the issue and `it` is running through using nearly all CS rules. i had to disable a handful of them due to not finding an easy fix for the test project and, obviously, all template rules are not activated either. if we want to activate them, we need to execute API requests for each of them with a valid configuration.
however, i can also confirm that the build breaks if either the `sonar:sonar` target fails or a CS rule is violated.

i am just not sure why wercker is failing. i saw that some quality gate is not passed on sonarcloud.io but if i check the branch's quality gate evaluation, sonarcloud.io actually says `Passed`.

anyhow, now `RunPluginTest` violates one PMD rule which i am not sure how to fix it as we simply require so many imports. can you disable this rule or weaken it s.t. the number of allowed imports is set to a higher value?
once this is fixed, we could merge it into `master`?

and thank you for speeding up the testing on travis with a cached version of SQ as well as the travis integration. :)

romani commented:
>[INFO] PMD Failure: org.checkstyle.plugins.sonar.RunPluginTest:1 Rule:ExcessiveImports Priority:3 A high number of imports can indicate a high degree of coupling within an object..

Please suppress this violation in pmd config, with comment, we value incapsulation of all in one test more than high coupling.

>once wercker and pmd are fixed, we could merge it into master?

Yes, please fix issues from inspections also.
I will review sonarqube , it might be because I switched gate to demand zero violation in whole project (a levey that we get in main project) as gate is the same. I can create new gate for sonar plugin to check only new code while plugin is fighting for better code.

Please address all other, sonarqube is on me.

muhlba91 commented:
PMD seems to be fixed now and IDEA is also running through.

I then faced an issue with findbugs and JDK 11 but using SpotBugs instead solved this.

Now we have got another problem: cobertura & JDK 11. cobertura doesn't support JDK 11 and all builds break because of getting run with JDK 11 (no matter whether we specify `1.8` or `11` in the `pom.xml`) and JaCoCo is the only supported coverage tool for JDK 9+.
I think a general conversion to JaCoCo might be not too difficult (hopefully) but I have no idea whether the current cobertura configuration with `regexes` to support coverage metrics per file is supported by JaCoCo or if we have to lower the general criteria for the entire code base here.
@romani can you help here? Or should we try replacing it and setting the thresholds just so that it passes with our current code base?

romani commented:
>Or should we try replacing it and setting the thresholds just so that it passes with our current code base?

Yes, just replace

romani commented:
This PR become too big and more and more updates jumping on this train .... so it will be hard to keep it on leash.

I think we need to split this PR in multiple in sequence and do not keep that much changes in active review:
1) update to spotbugs
2) create ITs base on orchestrator  and sonar 6.7 - https://repox.jfrog.io/repox/sonarsource/org/sonarsource/sonarqube/sonar-application/6.7.7/sonar-application-6.7.7.zip , make ITs to be executed only at CI by default to not bother local with 200Mb download.
3) update to jacoco to let sonarqube.io see such statistics.
4) release update to checkstyle 8.28
5) do release 
6) update to jdk 11
7) actual update to 7.9 Sonar

muhlba91 commented:
i agree, with all those force pushes it‘s getting more and more.

however, it seems only jacoco is missing now and we adapted the IT to 7.9 API which is why i‘d actually suggest either:
1)
- add usage of jacoco in another PR,
- extract jdk 11 update from here,
- finish this one off with the IT and SQ update together,
- update to 8.28, or
2)
- update to 8.28,
- update jacoco,
- update jdk 11 -> could be combined with previous one though,
- update to 7.9,
- introduce IT (or combine with previous).

i wouldn‘t add the IT before the SQ 7.9 update as we know it‘s running through now and we should avoid adapting it to 6.7 & JDK 8 and then apply the changes again for JDK 11 & SQ 7.9 or run into troubles to get it running. sure, the changes could be minor but i don‘t know atm and would like not to play around and waste time if we know the SQ update follows anyways shortly after.

also, i‘d not mix the CS 8.28 update to somewhere into the middle as we then create dependencies which might delay it more; hence, my suggestion of doing it immediately now or waiting after until the SQ update.

i cannot judge how much time pressure the CS 8.28 update gives us in terms of necessary awaited features; though, personally, i‘d rather try finishing this one off and then do the update, maybe skipping one release to update directly to the soon upcoming 8.29, if it‘s a feasible possibility.

romani commented:
I did few PRs to make update granular ... 

> i wouldn‘t add the IT before the SQ 7.9 update as we know it‘s running through now and we should avoid adapting it to 6.7 & JDK 8 

ok, but make sense to try ... to keep it clear what for we need each change, updates like "all at ones" ... never end up in good way. In future it is hard to get to know why it was done.

Lets move jacoco to separate PR for sure, there might be bunch nuances.

> sonarqube is on me.

new gate is created https://sonarcloud.io/organizations/checkstyle/quality_gates/show/9510 it should not make unrelated problems

romani commented:
I failed to quickly run orchestrator on 6.7.7
```
> 2020.01.19 17:48:51 INFO  ce[][o.s.c.c.CePluginRepository] Loaded plugin Orchestrator :: Reset data [resetdata]
> 2020.01.19 17:48:52 INFO  ce[][o.s.c.q.PurgeCeActivities] Delete the Compute Engine tasks created before Tue Jul 23 17:48:52 PDT 2019
> 2020.01.19 17:48:52 INFO  ce[][o.s.c.q.PurgeCeActivities] Delete the Scanner contexts tasks created before Sun Dec 22 17:48:52 PST 2019
> 2020.01.19 17:48:52 INFO  ce[][o.s.ce.app.CeServer] Compute Engine is operational
> 2020.01.19 17:48:52 INFO  app[][o.s.a.SchedulerImpl] Process[ce] is up
> 2020.01.19 17:48:52 INFO  app[][o.s.a.SchedulerImpl] SonarQube is up

java.lang.AssertionError: Failed to execute build.

	at org.junit.Assert.fail(Assert.java:88)
	at org.checkstyle.plugins.sonar.RunPluginTest.testSonarExecution(RunPluginTest.java:91)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)

> 2020.01.19 17:48:52 INFO  app[][o.s.a.SchedulerImpl] Stopping SonarQube
> 2020.01.19 17:48:53 INFO  ce[][o.s.p.StopWatcher] Stopping process
> 2020.01.19 17:48:53 INFO  ce[][o.s.ce.app.CeServer] Compute Engine is stopping...
> 2020.01.19 17:48:53 INFO  ce[][o.s.c.t.CeProcessingSchedulerImpl] Waiting for workers to finish in-progress tasks
> 2020.01.19 17:48:54 ERROR ce[][c.g.c.u.c.ExecutionList] RuntimeException while executing runnable com.google.common.util.concurrent.Futures$6@453ffd1c with executor org.sonar.ce.taskprocessor.CeProcessingSchedulerExecutorServiceImpl@32fe8e11
> java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@14ffc044 rejected from java.util.concurrent.ScheduledThreadPoolExecutor@601869d6[Shutting down, pool size = 1, active threads = 1, queued tasks = 0, completed tasks = 0]
> 	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2047)
> 	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:823)
> 	at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:326)
> 	at java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:533)
> 	at java.util.concurrent.ScheduledThreadPoolExecutor.execute(ScheduledThreadPoolExecutor.java:622)
> 	at com.google.common.util.concurrent.MoreExecutors$ListeningDecorator.execute(MoreExecutors.java:550)
> 	at org.sonar.server.util.AbstractStoppableExecutorService.execute(AbstractStoppableExecutorService.java:133)
> 	at com.google.common.util.concurrent.ExecutionList.executeListener(ExecutionList.java:156)
> 	at com.google.common.util.concurrent.ExecutionList.execute(ExecutionList.java:145)
> 	at com.google.common.util.concurrent.ListenableFutureTask.done(ListenableFutureTask.java:91)
> 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:384)
> 	at java.util.concurrent.FutureTask.set(FutureTask.java:233)
> 	at java.util.concurrent.FutureTask.run(FutureTask.java:274)
> 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
> 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
> 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
> 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
> 	at java.lang.Thread.run(Thread.java:745)
> 2020.01.19 17:48:55 INFO  ce[][o.s.ce.app.CeServer] Compute Engine is stopped
> 2020.01.19 17:48:56 INFO  app[][o.s.a.SchedulerImpl] Process [ce] is stopped
> 2020.01.19 17:48:57 INFO  web[][o.s.p.StopWatcher] Stopping process
> 2020.01.19 17:48:59 INFO  app[][o.s.a.SchedulerImpl] Process [web] is stopped
> 2020.01.19 17:49:00 WARN  app[][o.s.a.p.AbstractProcessMonitor] Process exited with exit value [es]: 143
> 2020.01.19 17:49:00 INFO  app[][o.s.a.SchedulerImpl] Process [es] is stopped
> 2020.01.19 17:49:00 INFO  app[][o.s.a.SchedulerImpl] SonarQube is stopped
```

as it working on 7.9 .... good we will introduce it together with update to 7.9

romani commented on pom.xml:
```diff
@@ -528,8 +592,24 @@
+            <configuration>
+              <rules>
+                <requireJavaVersion>
+                  <version>11</version>
```
after rebase please use here `java.version`

romani commented on src/it/java/org/checkstyle/plugins/sonar/RunPluginTest.java:
```diff
@@ -0,0 +1,260 @@
+    public static void beforeAll() {
+        orchestrator = Orchestrator.builderEnv()
+                .setZipFile(new File("target/temp-downloads/sonar-application-7.9.2.zip"))
+                //.setSonarVersion(System.getProperty("sonar.runtimeVersion", "7.9.2"))
```
```
private static final String SONAR_APP_VESION = "7.9.2";

                .setZipFile(new File("target/temp-downloads/sonar-application-"
                        + SONAR_APP_VESION + ".zip"))
                //.setSonarVersion(System.getProperty("sonar.runtimeVersion", SONAR_APP_VESION))
```

romani commented on src/it/java/org/checkstyle/plugins/sonar/RunPluginTest.java:
```diff
@@ -0,0 +1,260 @@
+                .addPlugin(FileLocation.byWildcardMavenFilename(new File("target"),
+                        "checkstyle-sonar-plugin-*.jar"))
+                .addPlugin(MavenLocation.of("org.sonarsource.sonar-lits-plugin",
+                        "sonar-lits-plugin",
```
why we need this plugin ?

muhlba91 replied:
i actually didn't find much documentation about it but from playing around with the test project and all rules activated it actually finds out the issues found in the project, if none found the file is empty and, hence, the build passes.
i thought in the end it makes sense to make sure the test project actually passes all activated rules.

romani replied:
ok, let it be for some time here , but better to not test not our plugins.

romani commented on src/it/java/org/checkstyle/plugins/sonar/RunPluginTest.java:
```diff
@@ -0,0 +1,260 @@
+        }
+
+        // copy project to analysis space
+        Path projectRoot = Paths.get("test-resources/" + PROJECT_NAME);
```
please move project that we need for ITs to `it` resources
`Path projectRoot = Paths.get("it/java/resources/" + PROJECT_NAME);`

romani replied:
Does it make sense to move resources required for `it` under such folder ? We split tests by folders, the same is good to do for resources

muhlba91 replied:
yes sure, i actually moved them already but it seems this commit got lost or i didn't finish committing it - sorry for that!
i moved them to `it/resources/`

romani commented on src/main/java/org/sonar/plugins/checkstyle/CheckstyleProfileExporter.java:
```diff
@@ -224,7 +226,7 @@ private static void appendRuleParameters(Writer writer, ActiveRuleWrapper active
 
     private static void appendModuleProperty(Writer writer, String propertyKey,
-            String propertyValue) throws IOException {
+            @Nullable String propertyValue) throws IOException {
```
what tool ask you to do this ?
please move to separate commit if not related to 7.9

muhlba91 replied:
IDEA when updating to JDK 11

romani commented on src/test/java/org/sonar/plugins/checkstyle/internal/ChecksTest.java:
```diff
@@ -189,7 +189,7 @@ private static void validateSonarRuleProperties(Class<?> module, Set<Node> param
 
             Assert.assertFalse(moduleName
-                    + " requires a valid key for unknown parameter in sonar rules"
+                            + " requires a valid key for unknown parameter in sonar rules"
```
revert such formatting or move to separate commit.
same for all other like this

muhlba91 commented:
@romani #267 contains usage of jacoco, #268 contains update to JDK 11. please merge in order. ;)
once both are merged, we can continue with this one.

romani commented:
Please rebase on latest master

muhlba91 commented:
@romani all running through now

romani commented:
> contains usage of jacoco,

this will be continued at #270

romani commented:
I merged this PR, few issues are better to resolve separately.
